modules/planning/tasks/deciders/path_bounds_decider.cc

```diff
@@ -16,19 +16,174 @@
     : Decider(config) {}
 
 Status PathBoundsDecider::Process(
     Frame *const frame, ReferenceLineInfo *const reference_line_info) {
-  // TODO(all) implement
+  // Sanity checks.
+  CHECK_NOTNULL(frame);
+  CHECK_NOTNULL(reference_line_info);
+
+  // The decided path bounds should be in the format of: (s, l_min, l_max).
+  std::vector<std::tuple<double, double, double>> path_boundaries;
+
+  // 0. Initialize the path boundaries to be an indefinitely large area.
+  if (!InitPathBoundaries(
+          reference_line_info->reference_line(),
+          frame->PlanningStartPoint(), &path_boundaries)) {
+    const std::string msg = "Failed to initialize path boundaries.";
+    AERROR << msg;
+    return Status(ErrorCode::PLANNING_ERROR, msg);
+  }
+  // 1. Decide a rough boundary based on road info and ADC's position
+  if (!GetBoundariesFromRoadsAndADC(
+          reference_line_info->reference_line(),
+          reference_line_info->AdcSlBoundary(), &path_boundaries)) {
+    const std::string msg = "Failed to decide a rough boundary based on "
+                            "road information.";
+    AERROR << msg;
+    return Status(ErrorCode::PLANNING_ERROR, msg);
+  }
+  // 2. Finetune the boundary based on static obstacles
+  // TODO(all): in the future, add side-pass functionality.
+  if (!GetBoundariesFromStaticObstacles(
+          reference_line_info->path_decision()->obstacles(),
+          &path_boundaries)) {
+    const std::string msg = "Failed to decide fine tune the boundaries after "
+                            "taking into consideration all static obstacles.";
+    AERROR << msg;
+    return Status(ErrorCode::PLANNING_ERROR, msg);
+  }
+
+  // Update the path boundary info to the frame.
+  // TODO(all): update frame with path bound info.
   return Status::OK();
 }
 
+bool PathBoundsDecider::InitPathBoundaries(
+    const ReferenceLine& reference_line,
+    const common::TrajectoryPoint& planning_start_point,
+    std::vector<std::tuple<double, double, double>>* const path_boundaries) {
+  // Sanity checks.
+  CHECK_NOTNULL(path_boundaries);
+  path_boundaries->clear();
+
+  // Starting from ADC's current position, increment until the horizon, and
+  // set lateral bounds to be infinite at every spot.
+  common::FrenetFramePoint adc_frenet_position = reference_line.GetFrenetPoint(
+      planning_start_point.path_point());
+  for (double curr_s = adc_frenet_position.s();
+       curr_s < std::min(adc_frenet_position.s() + kPathBoundsDeciderHorizon,
+                         reference_line.Length());
+       curr_s += kPathBoundsDeciderResolution) {
+    std::tuple<double, double, double> curr_bound = std::make_tuple(
+        curr_s,
+        std::numeric_limits<double>::lowest(),
+        std::numeric_limits<double>::max());
+    path_boundaries->emplace_back(curr_bound);
+  }
+
+  return true;
+}
+
+bool PathBoundsDecider::GetBoundariesFromRoadsAndADC(
+    const ReferenceLine& reference_line,
```

Comment:
Return value issue. Same here.

Reply:
Sometimes, when there is some failure inside the function, we may want to return false. For example, line 132 and line 144 currently output warning msg and use some temporary remedy, but later we may change them to "return false".